ektogamat/R3F-Ultimate-Lens-Flare
# Packaging setup

## Description

This PR setups up:

- `esbuild` to bundle `@andersonmancini/lens-flare` and eventually ship it to NPM (version script still need to be done)
-  A new example using Next.js that uses `@andersonmancini/lens-flare` as a package
- A small monorepo using [Turborepo](https://turbo.build/repo) so we can easily:
   1. Spin up a dev environment: this will start the example in dev mode along with `@andersonmancini/lens-flare` so it's easy to update the package against a real life example
   2. Build both the example and the package to deploy it to Vercel
   3. Extend the pipeline in the future if we want to add other tools (eslint, prettier, tests, etc.)
   
 
Remaining tasks to do:

- [x] Add versioning and publish script to `@andersonmancini/lens-flare` 
- [x] Fix issues with `wrapEffect` not being defined in the bundled code
- [x] Add deploy script for example: add `vercel` back as a dev dependency
- [x] Set up esbuild watcher for dev
- [x] Update README with new instructions


## How to test?

- Pull this branch
- Make sure to remove any previously installed node_module (I recommend running `sudo rm -rf node_modules`)
- Install the dependencies from the root of the project with `npm install`. This will install the dependency for both `@andersonmancini/lens-flare` and the example project
- Run `turbo dev` to start both `@andersonmancini/lens-flare` and the example project in dev mode. Head to `http://localhost:3000` and verify that everything is working.
- Run `turbo start` to build both `@anderson/lens-flare` and the example and serve the resulting build at `http://localhost:3000`. Verify that everything is working as expected.

I can jam with you to run through how to publish all this on NPM and do a final walk through/release next week!

🙂
